pkg/operator/controllers/dnsmasq/scripts/99-dnsmasq-restart.gotmpl

```diff
@@ -7,12 +7,43 @@
 interface=$1
 action=$2
 
+main() {
+    if [[ $interface == eth* && $action == "up" ]] || [[ $interface == eth* && $action == "down" ]] || [[ $interface == enP* && $action == "up" ]] || [[ $interface == enP* && $action == "down" ]]; then
+        log "$action happened on $interface, connection state is now $CONNECTIVITY_STATE"
+        sanity_check
```

Comment:
Super duper nit, but we might want to just call it what it is to avoid some jumping around in the code:
```suggestion
        check_resolv_files
```
Or something along those lines.